test/cases/global_identification_test.rb

```diff
@@ -19,4 +19,22 @@ class GlobalIdentificationTest < ActiveSupport::TestCase
     assert_equal SignedGlobalID.create(@model, for: 'login'), @model.to_signed_global_id(for: 'login')
     assert_equal SignedGlobalID.create(@model, for: 'login'), @model.to_sgid(for: 'login')
   end
+
+  test 'raises a friendly error for blank ids' do
+    message = 'Unable to create a GlobalID for Person without an id. ' \
+              '(Did you forget to call save?)'
+    person = Person.new(nil)
```

Comment:
`@model` which is a `PersonModel` has a settable id. Change to `@model.id = nil` and reuse `@model` here.